library/src/main/java/com/schibsted/spain/barista/assertion/BaristaHintAssertions.kt

```diff
@@ -2,20 +2,49 @@ package com.schibsted.spain.barista.assertion
 
 import android.support.annotation.IdRes
 import android.support.annotation.StringRes
-import android.support.test.espresso.matcher.ViewMatchers.withHint
+import android.support.design.widget.TextInputEditText
+import android.support.design.widget.TextInputLayout
+import android.support.test.InstrumentationRegistry
 import android.support.test.espresso.matcher.ViewMatchers.withId
+import android.view.View
+import android.widget.FrameLayout
+import android.widget.TextView
 import com.schibsted.spain.barista.internal.assertAny
+import org.hamcrest.Description
+import org.hamcrest.Matcher
+import org.hamcrest.TypeSafeMatcher
 
 object BaristaHintAssertions {
 
     @JvmStatic
     fun assertHint(@IdRes viewId: Int, @StringRes text: Int) {
-        withId(viewId).assertAny(withHint(text))
+        val string = InstrumentationRegistry.getTargetContext().resources.getString(text)
+        withId(viewId).assertAny(withLayoutHint(string))
     }
 
     @JvmStatic
     fun assertHint(@IdRes viewId: Int, text: String) {
-        withId(viewId).assertAny(withHint(text))
+        withId(viewId).assertAny(withLayoutHint(text))
     }
 
+    private fun withLayoutHint(expectedHint: String): Matcher<View> {
```

Comment:
I think we may think in a better name for this method... but I can't propose any 😅